cmd/createtree/main.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/google/trillian"
+	"github.com/google/trillian/crypto/sigpb"
+	"google.golang.org/grpc"
+)
+
+var (
+	adminEndpoint = flag.String("admin_endpoint", "", "Endpoint of the gRPC Trillian Admin Server")
+
+	treeState          = flag.String("tree_state", trillian.TreeState_ACTIVE.String(), "State of the new tree")
+	treeType           = flag.String("tree_type", trillian.TreeType_LOG.String(), "Type of the new tree")
+	hashStrategy       = flag.String("hash_strategy", trillian.HashStrategy_RFC_6962.String(), "Hash strategy (aka preimage protection) of the new tree")
+	hashAlgorithm      = flag.String("hash_algorithm", sigpb.DigitallySigned_SHA256.String(), "Hash algorithm of the new tree")
+	signatureAlgorithm = flag.String("signature_algorithm", sigpb.DigitallySigned_RSA.String(), "Signature algorithm of the new tree")
+	duplicatePolicy    = flag.String("duplicate_policy", trillian.DuplicatePolicy_DUPLICATES_NOT_ALLOWED.String(), "Duplicate policy of the new tree")
+	displayName        = flag.String("display_name", "", "Display name of the new tree")
+	description        = flag.String("description", "", "Description of the new tree")
+
+	privateKeyType = flag.String("private_key_type", "PEMKeyFile", "Type of private key to be used")
```

Comment:
Renamed to `--private_key_format`. I've kept the default as `PEMKeyFile`, though, so it matches the proto used on the RPC.